src/rust/engine/graph/src/lib.rs

```diff
@@ -1007,6 +1009,45 @@ impl<N: Node> Graph<N> {
       Ok(())
     }
   }
+
+}
+
+// This module provides a trait which contains functions that 
+// should only be used in tests. A user must explicitly import the trait
+// to use the extra test functions, and they should only be imported into 
+// test modules.
+pub mod test_support {
+  use super::{EntryState, EntryId, Node};
+  pub trait TestGraph<N: Node> {
+    fn set_fixture_entry_state_for_id(&self, id: EntryId, state: EntryState<N>);
+    fn add_fixture_entry(&self, node: N) -> EntryId;
+    fn entry_state(&self, id: EntryId) -> &str;
+  }
+}
+
+impl<N: Node> test_support::TestGraph<N> for Graph<N> {
```

Comment:
Can this be inside the `test_support` module? And if so, can that module be marked `#[cfg(test)]`?

Reply:
Unfortunately the compiler doesn't know to match the #[cfg(test)]  and it doesn't propagate between modules both declared that way. So the compiler thinks that the test_support module doesn't exist when it's building the watch_tests, which import it.

I did move the impl into the module which is more hygienic.